src/main/java/com/chatapp/vartalaab/service/UserSessionService.java

```diff
@@ -24,4 +25,25 @@ public void saveUserSessionDetails(UserSessionDetails userSessionDetails){
     public Optional<UserSessionDetails> getUserSessionDetails(String username){
         return userSessionRepository.findById(username  + GeneralUtility.SESSION_APPENDER);
     }
+
+    public void updateUserWebSocketSessionDetails(String username, String sessionId, boolean shouldAdd){
+        Optional<UserSessionDetails> userSession = this.getUserSessionDetails(username);
+        if(shouldAdd){
+            userSession.get().getWebSocketSessionIds().add(sessionId);
+        }
+        else{
+            userSession.get().getWebSocketSessionIds().remove(sessionId);
+        }
+        userSessionRepository.save(userSession.get());
+    }
+
+    public boolean checkIfUserIsOnline(String username){
```

Comment:
the function can be renamed to something smaller -> for ex: isUserOnline

Reply:
done